soc/alif/ensemble/common/startup.S

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2025 Alif Semiconductor.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Reset hook module for Alif Semiconductor ensemble family
+ */
+
+#include <zephyr/toolchain.h>
+#include <zephyr/linker/sections.h>
+
+_ASM_FILE_PROLOGUE
+
+GTEXT(SystemInit)
+GTEXT(soc_reset_hook)
+GDATA(_vector_table)
+
+#ifdef CONFIG_SOC_RESET_HOOK
+
+SECTION_FUNC(TEXT, soc_reset_hook)
+
+	/* Reset CONTROL register */
+	movs.n r0, #0
+	msr CONTROL, r0
+	isb
+#if defined(CONFIG_CPU_CORTEX_M_HAS_SPLIM)
```

Comment:
Good, I assume this makes it possible that we don't need to enable CONFIG_INIT_ARCH_HW_AT_BOOT by default.